addons/sale_management/models/sale_order.py

```diff
@@ -156,7 +160,7 @@ def action_confirm(self):
 
```

Comment:
Btw, if the code before fixes your problem, that might be the solution we should apply in 14.0, instead of the `with_user(SUPERUSER_ID)` always applied before the `send_mail`, because it still makes sense when a salesman confirms a SO to send the mail as the salesman :thinking:

Reply:
:ok_hand: https://github.com/odoo/odoo/pull/86632 here is a PR for V14, (if all the runbot test are good) from what i could try in locale it seems ok